Automattic/jetpack
# Portfolio CPT: Ensure Jetpack Settings page UI is clear for WoA sites with portfolio theme support

Fixes https://github.com/Automattic/vulcan/issues/523

## Proposed changes:

* The issue was that on WoA sites there is a new behaviour - introduced intentionally here: https://github.com/Automattic/jetpack/pull/39431 - whereby if using a theme that supports Portfolios CPTs, it isn't possible to toggle Portfolios off.
* This PR adds a message to the toggle UI to explain this and makes the toggle untoggleable (on the Jetpack > Settings > Writing page)
* One option would also have been to create an option to store the information so it was available via the React-based admin page, but to try and minimize performance issues I opted for the conditional creation of a variable.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/523

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Follow the testing instructions here: https://github.com/Automattic/jetpack/pull/39268

On a WoA site:
* When activating a theme that supports the Portfolio CPT (Sketch, Illustrator, Blask), the toggle at Jetpack > Settings > Writing (`/wp-admin/admin.php?page=jetpack#/writing`) should be disabled in the on position. You should also see some additional text in the description, explaining that the theme supports Jetpack Portfolios.
* This change will only show if the PR is applied with the Jetpack plugin - if only applied on the WordPress.com Features plugin then behaviour would be the same as with trunk.


Testing on self-hosted, there should be no changes in comparison to testing with trunk.

Testing on Simple, there should be no changes in comparison to testing with trunk.


On WoA with theme support:
<img width="1028" alt="Screenshot 2024-09-24 at 14 44 08" src="https://github.com/user-attachments/assets/c6537662-06ed-4fba-8d37-22bba9e45ea6">
